include/natalie/object.hpp

```diff
@@ -292,8 +292,9 @@ class Object : public Cell {
 
     bool is_a(Env *, Value) const;
     bool respond_to(Env *, Value, bool = true);
-    bool respond_to_method(Env *, Value, Value) const;
-    bool respond_to_method(Env *, Value, bool) const;
+    bool respond_to_method(Env *, Value, Value);
+    bool respond_to_method(Env *, Value, bool);
+    bool respond_to_missing(Env *, Value, Value);
```

Comment:
I'm sad to lose the `const` here. Is there a way we can keep these methods const?

Reply:
I haven't really checked this code, but I remember having to remove const somewhere during a change because the code had to call another method that wasn't marked as const. It's pretty much turtles all the way down, which probably could have its own ticket

Reply:
Yep turtles... as I understand it we would have to at least make `Object::send` const?

Reply:
In this case we're calling `Object#send`, which it something that can modify the receiver, so it shouldn't be `const`, so this can't be `const` either.

Reply:
I think we could potentially avoid `Object::send` and use `Method::call` directly (we don't need to call #method_missing or raise NoMethodError in this context), but that's not `const` either.

Reply:
It looks possible to constify that one, but I guess we should consider that to be a future work. I think the added value of this change is bigger than the added value of adding some const correctness at more or less random code paths.

Reply:
OK, thanks for taking a look. This is fine for now.